File: examples/set_operations.py
from typing import Set

def test_set_creation() -> Set[int]:
    """Test basic set creation and operations"""
    # Create sets using literal syntax
    s1 = {1, 2, 3}
    s2 = {3, 4, 5}
    
    # Create empty set
    empty = set()
    
    return s1

def test_set_operators() -> Set[int]:
    """Test set operators"""
    s1 = {1, 2, 3}
    s2 = {3, 4, 5}
    
    # Union
    union = s1 | s2  # {1, 2, 3, 4, 5}
    
    # Intersection
    intersection = s1 & s2  # {3}
    
    # Difference
    diff = s1 - s2  # {1, 2}
    
    # Symmetric difference
    sym_diff = s1 ^ s2  # {1, 2, 4, 5}
    
    return union

def test_set_methods():
    """Test set mutation methods"""
    s = {1, 2, 3}
    
    # Add element
    s.add(4)
    
    # Remove element (raises KeyError if not found)
    s.remove(2)
    
    # Discard element (no error if not found)
    s.discard(5)
    
    return s

def test_set_comprehension() -> Set[int]:
    """Test set comprehension"""
    # Set comprehension
    squares = {x * x for x in range(5)}
    
    # With condition
    even_squares = {x * x for x in range(10) if x % 2 == 0}
    
    return even_squares

File: examples/set_operations.rs
use std::collections::HashSet;
#[doc = "Test basic set creation and operations"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_set_creation() -> std::collections::HashSet<i32> {
    let s1 = {
        let mut set = HashSet::new();
        set.insert(1);
        set.insert(2);
        set.insert(3);
        set
    };
    s1
}
#[doc = "Test set operators"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_set_operators() -> std::collections::HashSet<i32> {
    let s1 = {
        let mut set = HashSet::new();
        set.insert(1);
        set.insert(2);
        set.insert(3);
        set
    };
    let s2 = {
        let mut set = HashSet::new();
        set.insert(3);
        set.insert(4);
        set.insert(5);
        set
    };
    let _cse_temp_0 = s1
        .union(&s2)
        .cloned()
        .collect::<std::collections::HashSet<_>>();
    let union = _cse_temp_0;
    union
}
#[doc = "Test set mutation methods"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_set_methods() -> Vec<String> {
    let mut s = {
        let mut set = HashSet::new();
        set.insert(1);
        set.insert(2);
        set.insert(3);
        set
    };
    s.insert(4);
    if !s.remove(&2) {
        panic!("KeyError: element not in set")
    };
    s.remove(&5);
    s
}
#[doc = "Test set comprehension"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_set_comprehension() -> std::collections::HashSet<i32> {
    let even_squares = (0..10)
        .into_iter()
        .filter(|&x| x % 2 == 0)
        .map(|x| x * x)
        .collect::<HashSet<_>>();
    even_squares
}
